autopilot/followpoint.cpp

```diff
@@ -70,64 +75,81 @@ void FollowPoint::holdPosition()
         mMovementController->setDesiredSteering(0.0);
         mMovementController->setDesiredSpeed(0.0);
     } else {
-        mVehicleConnection->requestVelocityAndYaw({}, mVehicleConnection->getVehicleState()->getPosition(mPosTypeUsed).getYaw());
+        mVehicleConnection->requestGotoENU(getCurrentVehiclePosition().getXYZ(), true);
     }
 }
 
-void FollowPoint::setPosTypeUsed(const PosType &posTypeUsed)
+void FollowPoint::pointToFollowInVehicleFrame(const PosPoint &point)
 {
-    mPosTypeUsed = posTypeUsed;
+    if (pointIsNew(point)) {
+        // ToDo: Apply follow point parameters here
+        mLineFromVehicleToPoint.setP1(QPointF(0,0));
+        mLineFromVehicleToPoint.setP2(point.getPoint());
+        mDistanceToPointIn2D = mLineFromVehicleToPoint.length();
+    }
 }
 
-void FollowPoint::updateFollowPointInVehicleFrame(const PosPoint &point)
+void FollowPoint::pointToFollowInEnuFrame(const PosPoint &point)
 {
+    if (pointIsNew(point)) {
+        // ToDo: Apply follow point parameters here
+        mCurrentPointToFollowInEnuFrame = point;
+        mCurrentPointToFollowInEnuFrame.setHeight(point.getHeight() + mFollowPointHeight);
+
+        mDistanceToPointIn2D = getCurrentVehiclePosition().getDistanceTo(mCurrentPointToFollowInEnuFrame);
+    }
+}
+
+bool FollowPoint::pointIsNew(const PosPoint &point)
```

Comment:
The name suggests that it is a simple test without side effects but it can actually reset the timer and affect the whole state machine. I think updatePointToFollowInVehicleFrame/ updatePointToFollowInEnuFrame need to be refactored for clarity to sth like:

if (pointIsNew(point) {
  .. update state stuff
 if ((mStmState == FollowPointSTMstates::FOLLOWING || mStmState == FollowPointSTMstates::WAITING)
   resetTimer..
}